charts/jupyter-with-spark/README.md

```diff
@@ -24,33 +24,87 @@ This chart is work-in-progress
+     enableHistoryEvents: true
+   ```
+ 
+   2. Specify the location of event log directory on your GCS bucket in values.yaml.
+
+   ```python
+   # eventsLogDir should point to a URI of GCS bucket where history events will be dumped
+   eventLogDir: "gs://my_bucket_name/event_logs/"
+   ```
+
+   3. Uncomment the line in conf/spark/spark-defaults.conf to specify the keyfile for accessing your GCS bucket.
+
+   **NOTE:** You should only update the name of your keyfile, retaining its path to be /etc/secrets/.
+
+   ```python
+   # Uncomment below line and replace sparkonk8s-test.json with the actual name of your keyfile
+   # to enable access to Google Cloud Storage.
+   spark.hadoop.google.cloud.auth.service.account.json.keyfile   /etc/secrets/bucket-access-key.json
+   ```
+
+   4. Place your GCS keyfile under conf/secrets directory of the chart.
+
+Refer to the [readme doc](../spark-hs/README.md) of Spark History Server chart, to launch the history server and monitor event logs of all your Spark applications.
+
+### Setting password for your Jupyter notebook server
+This chart by default enables authentication for your Jupyter notebook server. The default password is abc123
+
+Change this password by setting it as the value of `jupyterService.password` attribute in values.yaml.
+
+```python
+jupyterService:
+  type: LoadBalancer
+  port: 8888
+  # Set your password to access the notebook server. A default ('abc123') has been set for you.
+  # Setting the password to empty string will disable the authentication (not recommended).
+  password: 'your-new-password'
+```
+
+**NOTE**: if you set a new password via conf/jupyter/jupyter_notebook_config.py, it'll still be overridden by what you specify in values.yaml.
 
 # Running Spark application via notebook 
-While creating the SparkContext, in addition to the usual conf settings required to launch Spark on kubernetes, user must
-also set the "spark.master" in SparkConf as given below, to be able to launch the driver and executors on the kubernetes cluster.
+Create a new notebook based on Python 2 and launch a Spark cluster by creating SparkContext.
+
+```python
+spark = SparkSession.builder.config("spark.app.name", "spark-pi")\
+      .config("spark.executor.instances", "2")\
+      .getOrCreate()
+```
+Note that, other spark configurations like `spark.master`, `spark.submit.deployMode`, docker image names for driver and executor pods, kubernetes namespace are already set for you.
+See spark-defaults.conf under conf/spark/ directory of the chart for more details. You can also override the conf properties or set new ones in the notebook.
 
 ```python
-conf.set("spark.master", "k8s://https://{}:{}".format(
-        os.getenv("KUBERNETES_SERVICE_HOST"),
-        os.getenv("KUBERNETES_SERVICE_PORT")
-      ))
+spark = SparkSession.builder.config("spark.app.name", "spark-pi")\
+      .config("spark.kubernetes.namespace", "dev-namespace")\
+      .config("spark.executor.instances", "5")\
```

Comment:
Can you keep this to 1 or 2 instances? else, by default, you won't have the resources to run. I have increased the PKS cluster to 5 workers, now.